config/parse_config.go

```diff
@@ -1,30 +1,73 @@
-
 package config
 
 import (
-	"os"
 	"fmt"
+	"os"
 
-	"io/ioutil"
 	"gopkg.in/yaml.v2"
+	"io/ioutil"
 )
 
+// configuration interface for printing
+type ConfigStruct interface {
+	PrintStruct()
+}
+
+// explicitly define the tags for each field
+type configuration struct {
```

Comment:
Just a side note. I would typically expect data structs to be exported by default. I can't find a citable rule off-hand, but in general I find go tends to err on the side of data visibility, rather than data encapsulation.